build-info-extractor-go/src/main/java/org/jfrog/build/extractor/go/extractor/GoPublish.java

```diff
@@ -118,10 +118,12 @@ private File archiveProjectDir() throws IOException {
 
         try (ZipOutputStream zos = new ZipOutputStream(new FileOutputStream(zipFile))) {
             List<Path> pathsList = Files.walk(path)
-                    .filter(p -> !Files.isDirectory(p) && !p.toFile().getName().equals(zipFile.getName()))
+                    // Remove .git dir content, directories and the temp zip file
+                    .filter(p -> !path.relativize(p).startsWith(".git/") && !Files.isDirectory(p) && !p.toFile().getName().equals(zipFile.getName()))
                     .collect(Collectors.toList());
             for (Path filePath : pathsList) {
-                ZipEntry zipEntry = new ZipEntry(path.relativize(filePath).toString());
+                // We need to have the parent hierarchy of the project in order to use Artifactory's packing algorithm
```

Comment:
Instead of "Artifactory's packing algorithm" I would say "the zip packaging code in org.jfrog.build.extractor.go.extractor.GoZipBallStreamer".